### ISSUE TITLE
Generated .json file includes the full path of .graphql file

### ISSUE THREAD

**Issue by vjcagay** [apollographql/graphql-document-collector#9]
Let's say I have `User.graphql` located in `src/client/ts/graphql/queries/`. When I run `graphql-document-collector` it generates a `document.json` file that contains:

```
{
   "src/client/ts/graphql/queries/User.graphql" : "..."
}
```
Is there any way `graphql-document-collector` just generate `User.graphql` without including the full path? I have to put the `.graphql` files in the root directory just to avoid this kind of output.

**Comment by olaf89** [comment]
I had the same problem, i dont think it was like this always. In the past im almost sure this lib was working as you described.

As a work around im getting documents using helper function instead of by key,

import { pathEq, values } from 'ramda';
const graphqlDocs = require('../../documents.json');

function getDocument(docName): any {
    const allDocs = values(graphqlDocs);
    return allDocs.find(pathEq(['name', 'value'], docName));
}

**Comment by vjcagay** [comment]
In my case I just created a function that appends the full path of the file before executing the query:

```typescript
// graphqlhelper.ts
const query = (GraphQLFile: string, variables: { [key: string]: any } = {}) => {
  return apolloClient.query({
    query: docs[`src/client/ts/graphql/queries/${GraphQLFile}.graphql`],
    variables,
  });
};

export { query };
```

Then I could just call it by passing the file name, for instance, `User.graphql`:

```typescript
// somefile.ts
import { UserQuery } from './graphql/schema';
import { query } from './graphql/graphqlhelper';

query('User').then(({ data}: { data: UserQuery }) => { ... })
```

This works fine at the moment, but it would be nice not having to do this workaround.

**Comment by vjcagay** [comment]
@olaf89 Since the `.json` file generated contains the full path, it would make the application less portable, since you mentioned you worked on a Mac. I am working in Linux, and I'm getting the same result. If I want to make it portable I need to add a code that would detect the operating system and then apply the necessary string to append. This is making more complicated than it needs to be.

**Comment by olaf89** [comment]
@vjcagay are you referring to my "fix"? Cause overall i think its bug in this repo and should be solved, so at least the result in the same on every env. If you are referring to my fix Im not using key, instead im going into each document object and getting ".name.value" which is name of the query without path. I think it should work on every env.

**Comment by vjcagay** [comment]
@olaf89 no, I was not referring to your "fix". Actually I'm considering going the same way you did because I think that approach is better than mine. 

Yes I do think this is definitely a bug on this repo.

**Comment by olaf89** [comment]
Well Im using also some benefits using it this way even when this bug is fixed, first its not depending on  graphql-document-collector lib, so you have only a function that returns doc by name, and second you dont need to each time provide ".graphql" if you append it inside this function.

**Comment by Stradivario** [comment]
Maybe it is an old issue but let me share my experience:

I am collecting my documents using `graphql-document-collector` this beautiful program creates JSON file i put it inside ts file so it will be easily imported from typescript using `export const DOCUMENTS = {}`
Then i create based on this file mapping for every query mutation subscription:

`documentTypes.ts`

```
function strEnum<T extends string>(o: Array<T>): {[K in T]: K} {
    return o.reduce((res, key) => {
        res[key] = key;
        return res;
    }, Object.create(null));
}
export const DocumentTypes = strEnum(['change-password.graphql',
'add-activity.mutation.graphql',
'remove-activity.mutation.graphql',
'get-activities.query.graphql',
'login-with-linkedin.graphql',
'register.mutation.graphql',
'set-user-type.graphql',
'checkTokenValidity.graphql',
'add-quest-bulk-consultant.mutation.graphql',
'update-profile-state.mutation.graphql',
'listQuizPages.query.graphql']);
export type DocumentTypes = keyof typeof DocumentTypes;
```

`importQuery` helper

```typescript

import { DOCUMENTS } from './documents';
import { DocumentTypes } from './documentTypes';

export function importQuery(search: DocumentTypes) {
    let result;
    Object.keys(DOCUMENTS)
    .filter(doc => {
      if (doc.indexOf(<any>search) !== -1) {
        result = DOCUMENTS[doc];
      }
    });
    if (!result) {
      throw new Error(`Missing query: ${search}`);
    }
    return result;
}
```

Searching for correct query and returning result for it.

How to use:


```typescript

import { Injectable } from '@angular/core';
import { Apollo } from 'apollo-angular';
import { importQuery } from '../../modules/graphql/graphql.helpers';
import { IValidityType } from '../../modules/graphql';
import { Observable } from 'rxjs/Observable';
import 'rxjs/operators/map';

export const CHECK_VALIDITY_QUERY = importQuery('checkTokenValidity.graphql'');

@Injectable()
export class ConsultantService {

    constructor(
        private apollo: Apollo,
    ) { }

    listConsultants(token: string): Observable<IValidityType> {
        return this.apollo.query<IQuery>({
            query: CHECK_VALIDITY_QUERY,
            variables: { token }
        }).pipe(
        map(({ data }) => data.checkTokenValidity)
      );
    }

}
```


Regards to all of you for the good work creating this magnificent tool! :)
